### ISSUE TITLE
Load source model from a formatted file

### ISSUE THREAD

**Issue by dimembermatt** [lhr-solar/MPPT#12]
This is an issue that requires a little more in depth knowledge of the working code base.

Essentially, we want to enable multicell/multimodule source models, but to manually configure it by hand is repetitive and annoying. Therefore, it would be preferable to load it from a specific configuration file every time.

This configuration file should include relevant data about the following:
* the identifying test name (what am I emulating? A single cell? A subarray?)
* a description of the test
* the number of modules that should be chained together (we assume in series and shake in fear at the suggestion of wiring any of these in parallel)
* and a list of modules, in order, with their name, type, and environmental regime.

The environmental regime enumerates on what conditions (irradiance, temperature, eventually load) that the array is experiencing at specific moments. While we would expect adjacent modules to have environmental conditions continuous between them, the system will work without. Additionally, our source can interpolate data between given points, which is very handy.

An example of one such file, the trivial single cell, is presented below:
```
{
    "name": "Single Cell",
    "description": "A single cell. The default model with a pre-generated environmental regime.",
    "num_modules": 1,
    "pv_model": [
        {
            "module_num": 0,
            "module_type": "1x1",
            "env_regime": [
                [0, 0, 25],
                [1, 0, 25],
                [2, 0, 25],
                [3, 0, 25],
                [4, 0, 25],
                [5, 1000, 25],
                [10, 1000, 20],
                [20, 1000, 35],
                [30, 1000, 40],
                [40, 1000, 45],
                [50, 1000, 50],
                [60, 1000, 55],
                [70, 1000, 60],
                [80, 1000, 65],
                [85, 1000, 73.75],
                [90, 1000, 82],
                [95, 1000, 91.25],
                [100, 1000, 100],
                [110, 1000, 97.5],
                [120, 1000, 95],
                [130, 1000, 82.5],
                [140, 1000, 70],
                [150, 1000, 80],
                [160, 1000, 90],
                [165, 1000, 72.5],
                [170, 1000, 55],
                [175, 1000, 37.5],
                [180, 1000, 20],
                [185, 1000, 50],
                [190, 1000, 80],
                [195, 1000, 110],
                [200, 1000, 140]
            ] 
        }
    ]
}
```

We should constrain these models to specific rules in order to reflect the actual system:
* all modules are expected to have a bypass diode across it
* all modules listed are in series.

This post may be edited later with more constraints.

To recap, the task for this issue is to implement the function setup_f in source.py to accept and extract a file provided to it important details that will allow it to generate appropriate model output.

Specifically, we want to populate the following variables:
* name = String
* modules = [(module_type, 

A separate issue will be filed and linked

**Comment by dimembermatt** [comment]
Creation of a Cell class for the base unit, this is piped through the Source class with the same methods to accumulate module level and array level info.

**Comment by dimembermatt** [comment]
This is progressing very quickly in branch `mppt_simulator_feature#12`.
Still working out the quirks refactoring it out to cover expected test cases. main.py also needs to be updated.

**Comment by dimembermatt** [comment]
Current bugs that needs to be worked out:
- [ ] MPPT does not scale estimate to that of the source, for some of the algorithms we need to create a good estimate (probably coming from the source class about where optimal mpp is and pipe that into the algorithm.
- [ ] source is returning the last module's env_conditions on iterate; we should return a set of all of the env conditions and use those in our simulation class instead. Therefore we're going to need to figure out issue #8. We could potentially make a fifth graph and move all irradiance/temp series into a scatter there.

**Comment by dimembermatt** [comment]
This feature branch will contains changes for #6 and #8 as well.
